### PR TITLE
chore: change logs for 4.4.17

### PR DESCRIPTION
**Pull request by terry-xiaoyu**

### CODE REVIEW

File: zh_CN/changes/changes-4.4.md [hunk 1]
```diff
@@ -1,5 +1,52 @@
...
+  ```
+  [error] Accept error on 0.0.0.0:1883: emfile
+  ```
+  改进后的日志：
+  ```
+  [error] Accept error on 0.0.0.0:1883: EMFILE (Too many open files)
+  ```
+
+- 提升规则引擎在规则数量较多时的执行性能 [#10283](https://github.com/emqx/emqx/pull/10283)
+  在此改动之前，当规则数量比较多的时候，规则引擎的执行会成为瓶颈，规则引擎将耗费大量 CPU 在规则的查询和匹配上。
```
Comment (LenaLenaPan) [inline zh_CN/changes/changes-4.4.md]:
```suggestion
  改进前，当规则数量较多时，规则引擎将需要耗费大量 CPU 在规则的查询和匹配上，并成为性能瓶颈。
```

File: zh_CN/changes/changes-4.4.md [hunk 2]
```diff
@@ -1,5 +1,52 @@
...
+  本次优化中，通过简单地给规则列表添加一个缓存，大幅提升了此场景下的规则执行效率。
+  在我们的测试中，在一个 32 核 32G 的虚拟机上，我们创建了 700 条不执行任何动作的规则（绑定了 "do_nothing" 调试动作），
+  并以 1000 条每秒的速度向 EMQX 发送 MQTT 消息（即，规则触发频率为 700 * 1000 次每秒），
+  在应用此优化后，规则引擎的 CPU 使用率下降到了原来的 55% ~ 60%。
```
Comment (LenaLenaPan) [inline zh_CN/changes/changes-4.4.md]:
```suggestion
  在上述场景下，优化后的规则引擎 CPU 使用率下降到了之前的 55% ~ 60%。
```

File: zh_CN/changes/changes-4.4.md [hunk 3]
```diff
@@ -1,5 +1,52 @@
...
+
+### 修复
+
+- 修复 `Erlang distribution` 无法使用 TLS 的问题 [#9981](https://github.com/emqx/emqx/pull/9981)。
+  关于 `Erlang distribution`, 详见 [这里](https://www.emqx.io/docs/zh/v4.4/advanced/cluster.html)。
+
+- 修复 MQTT TLS 桥接无法验证对端的带通配符的证书的问题 [#10094](https://github.com/emqx/emqx/pull/10094)。
+
+- 修复 EMQX 由于大量 retain 消息而无法及时清除已掉线的 MQTT 连接信息的问题。[#10189](https://github.com/emqx/emqx/pull/10189)。
```
Comment (LenaLenaPan) [inline zh_CN/changes/changes-4.4.md]:
```suggestion
- 修复 EMQX 由于 retainer 中积压的消息过多，无法及时清除已掉线的 MQTT 连接信息的问题。[#10189](https://github.com/emqx/emqx/pull/10189)。
```

File: zh_CN/changes/changes-4.4.md [hunk 4]
```diff
@@ -1,5 +1,52 @@
...
+  在此修复之前，`emqx_retainer` 插件和 EMQX 的连接信息清理过程共用了同一个进程池，因此，
```
Comment (LenaLenaPan) [inline zh_CN/changes/changes-4.4.md]:
```suggestion
  修复前，`emqx_retainer` 插件和 EMQX 连接信息清理进程共用一个进程池，因此，
```

File: zh_CN/changes/changes-4.4.md [hunk 5]
```diff
@@ -1,5 +1,52 @@
...
+  如果该进程池被大量的 retain 消息下发任务阻塞时，许多已经掉线的 MQTT 连接将得不到及时清理。
+  详见 [#9409](https://github.com/emqx/emqx/issues/9409)。
+  此修复中，我们为 `emqx_retainer` 插件创建了单独的进程池，从而避免了该问题。
```
Comment (LenaLenaPan) [inline zh_CN/changes/changes-4.4.md]:
```suggestion
  修复后，EMQX 为 `emqx_retainer` 插件创建了单独的进程池，从而避免了该问题。
```

**Review by LenaLenaPan — COMMENTED** [review]
leave some comments

File: en_US/changes/changes-4.4.md [hunk 6]
```diff
@@ -1,5 +1,50 @@
...
+  ```
+  [error] Accept error on 0.0.0.0:1883: emfile
+  ```
+  Log messages after this change:
+  ```
+  [error] Accept error on 0.0.0.0:1883: EMFILE (Too many open files)
+  ```
+
+- Improve the performance of the rule engine when there are a large number of rules [#10283](https://github.com/emqx/emqx/pull/10283)
+  Before this change, when there were a large number of rules, the execution of the rule engine became a bottleneck, as the rule engine consumed a lot of CPU time on rule queries and matching.
```
Comment (LenaLenaPan) [inline en_US/changes/changes-4.4.md:27]:
```suggestion
Improved rule engine performance by implementing rule indexing to efficiently handle large rule sets and reduce CPU time on queries and matching.
```

File: en_US/changes/changes-4.4.md [hunk 7]
```diff
@@ -1,5 +1,50 @@
...
+### Bug fixes
+
+- Fix that `Erlang distribution` can't use TLS [#9981](https://github.com/emqx/emqx/pull/9981).
+  About `Erlang distribution`, See [here](https://www.emqx.io/docs/en/v4.4/advanced/cluster.html#distributed-erlang) for details.
+
+- Fixed MQTT bridge TLS connection could not verify wildcard certificate from peer [#10094](https://github.com/emqx/emqx/pull/10094).
+
+- Fixed the problem that EMQX could not timely clear the disconnected MQTT connection information due to a large number of retained messages [#10189](https://github.com/emqx/emqx/pull/10189).
+  Before this fix, the `emqx_retainer` plugin and the connection cleanup process of EMQX shared the same process pool.
+  Therefore, if the process pool was blocked by a large number of retain messages, many disconnected MQTT connections would not be cleaned up in time.
```
Comment (LenaLenaPan) [inline en_US/changes/changes-4.4.md]:
```suggestion
  Therefore, if the process pool was blocked by a large number of retain messages, many disconnected MQTT connections could not be cleaned up in time.
```

File: zh_CN/changes/changes-ee-4.4.md [hunk 8]
```diff
@@ -1,12 +1,58 @@
...
-## v4.4.16
+## e4.4.17
+
+*发布日期: 2023-04-13*
+
+### 增强
+
+- 改进从旧版本（4.2或更早）导入数据时的告警日志 [#1775](https://github.com/emqx/emqx-enterprise/pull/1775)。
+  在此改进之前，如果从旧版本（4.2及之前）的 EMQX 导出数据，并导入到4.4版本，内置认证部分的数据
+  可能会因为缺少认证类型而被忽略，但日志没有清楚地指出失败原因。
```
Comment (LenaLenaPan) [inline zh_CN/changes/changes-ee-4.4.md]:
```suggestion
  可能会因为缺少认证类型而被忽略，但用户无法从日志中清楚地获知失败原因。
```

**Review by LenaLenaPan — COMMENTED** [review]
add some comments

File: zh_CN/changes/changes-ee-4.4.md [hunk 9]
```diff
@@ -1,12 +1,58 @@
...
+  在此修改之后，将会提示用户使用命令行并指定认证类型导入：
```
Comment (LenaLenaPan) [inline zh_CN/changes/changes-ee-4.4.md]:
```suggestion
  在此修改之后，将会提示用户改用命令行工具做数据导入，并指定认证类型。
```

File: zh_CN/changes/changes-ee-4.4.md [hunk 10]
```diff
@@ -1,12 +1,58 @@
...
+## e4.4.17
+
+*发布日期: 2023-04-13*
+
+### 增强
+
+- 改进从旧版本（4.2或更早）导入数据时的告警日志 [#1775](https://github.com/emqx/emqx-enterprise/pull/1775)。
+  在此改进之前，如果从旧版本（4.2及之前）的 EMQX 导出数据，并导入到4.4版本，内置认证部分的数据
+  可能会因为缺少认证类型而被忽略，但用户无法从日志中清楚地获知失败原因。
+  在此修改之后，将会提示用户改用命令行工具做数据导入，并指定认证类型。
```
Comment (terry-xiaoyu) [inline zh_CN/changes/changes-ee-4.4.md]:
```suggestion
  在此修改之后，将会提示用户改用命令行工具做数据导入，并指定认证类型：
```

File: zh_CN/changes/changes-4.4.md [hunk 11]
```diff
@@ -1,5 +1,52 @@
 # 版本发布
 
+## v4.4.17
+
+*发布日期: 2023-04-13*
+
+### 增强
+
+- 启用了 `Proxy Protocol` 的监听器在收到 TCP 端口探测时，不再打印错误日志 [emqx/esockd#172](https://github.com/emqx/esockd/pull/172)。
+  修复前，如果监听器已启用了代理协议（`listener.tcp.external.proxy_protocol=on`），但连接在 TCP 握手完成后、接收到代理信息之前断开，则会打印以下错误日志：
```
Comment (terry-xiaoyu) [inline zh_CN/changes/changes-4.4.md:11]:
```suggestion

  修复前，如果监听器已启用了代理协议（`listener.tcp.external.proxy_protocol=on`），但连接在 TCP 握手完成后、接收到代理信息之前断开，则会打印以下错误日志：
```

File: zh_CN/changes/changes-4.4.md [hunk 12]
```diff
@@ -1,5 +1,52 @@
...
+  在我们的测试中，在一个 32 核 32G 的虚拟机上，我们创建了 700 条不执行任何动作的规则（绑定了 "do_nothing" 调试动作），
+  并以 1000 条每秒的速度向 EMQX 发送 MQTT 消息（即，规则触发频率为 700 * 1000 次每秒），
+  在上述场景下，优化后的规则引擎 CPU 使用率下降到了之前的 55% ~ 60%。
+
+### 修复
+
+- 修复 `Erlang distribution` 无法使用 TLS 的问题 [#9981](https://github.com/emqx/emqx/pull/9981)。
+  关于 `Erlang distribution`, 详见 [这里](https://www.emqx.io/docs/zh/v4.4/advanced/cluster.html)。
+
+- 修复 MQTT TLS 桥接无法验证对端的带通配符的证书的问题 [#10094](https://github.com/emqx/emqx/pull/10094)。
```
Comment (terry-xiaoyu) [inline zh_CN/changes/changes-4.4.md]:
```suggestion
- 修复 MQTT 桥接无法验证对端带通配符的 TLS 证书的问题 [#10094](https://github.com/emqx/emqx/pull/10094)。
```

File: zh_CN/changes/changes-4.4.md [hunk 13]
```diff
@@ -1,5 +1,52 @@
...
+
+- 修复 EMQX 由于 retainer 中积压的消息过多，无法及时清除已掉线的 MQTT 连接信息的问题。[#10189](https://github.com/emqx/emqx/pull/10189)。
```
Comment (terry-xiaoyu) [inline zh_CN/changes/changes-4.4.md]:
```suggestion
- 修复当 retainer 中积压的消息过多时，EMQX 无法及时清除已掉线的 MQTT 连接信息的问题。[#10189](https://github.com/emqx/emqx/pull/10189)。
```

File: zh_CN/changes/changes-4.4.md [hunk 14]
```diff
@@ -1,5 +1,52 @@
...
+  修复前，`emqx_retainer` 插件和 EMQX 连接信息清理进程共用一个进程池，因此，
```
Comment (terry-xiaoyu) [inline zh_CN/changes/changes-4.4.md]:
```suggestion
  修复前，`emqx_retainer` 插件和 EMQX 连接信息清理任务共用一个进程池，因此，
```

File: zh_CN/changes/changes-4.4.md [hunk 15]
```diff
@@ -1,5 +1,52 @@
...
+  如果该进程池被大量的 retain 消息下发任务阻塞时，许多已经掉线的 MQTT 连接将得不到及时清理。
```
Comment (terry-xiaoyu) [inline zh_CN/changes/changes-4.4.md]:
```suggestion
  如果该进程池被大量的 retain 消息下发任务阻塞时，许多已经掉线的 MQTT 连接信息将得不到及时清理。
```

File: zh_CN/changes/changes-4.4.md [hunk 16]
```diff
@@ -1,5 +1,52 @@
...
+  详见 [#9409](https://github.com/emqx/emqx/issues/9409)。
+  修复后，EMQX 为 `emqx_retainer` 插件创建了单独的进程池，从而避免了该问题。
```
Comment (terry-xiaoyu) [inline zh_CN/changes/changes-4.4.md]:
```suggestion
  修复后，`emqx_retainer` 插件使用单独的进程池，从而避免了该问题。
```

File: zh_CN/changes/changes-4.4.md [hunk 17]
```diff
@@ -1,5 +1,52 @@
...
+
+- 修复了 Helm Chart 中模板文件路径的错误。[#10229](https://github.com/emqx/emqx/pull/10229)
```
Comment (terry-xiaoyu) [inline zh_CN/changes/changes-4.4.md]:
```suggestion
- 修复了 Helm Chart 中模板文件 `service-monitor.yaml` 路径不正确的问题。[#10229](https://github.com/emqx/emqx/pull/10229)
```

File: zh_CN/changes/changes-ee-4.4.md [hunk 18]
```diff
@@ -1,12 +1,58 @@
 # 版本发布
 
-## v4.4.16
+## e4.4.17
+
+*发布日期: 2023-04-13*
+
+### 增强
+
+- 改进从旧版本（4.2或更早）导入数据时的告警日志 [#1775](https://github.com/emqx/emqx-enterprise/pull/1775)。
```
Comment (terry-xiaoyu) [inline zh_CN/changes/changes-ee-4.4.md]: remove the link

File: zh_CN/changes/changes-ee-4.4.md [hunk 19]
```diff
@@ -1,12 +1,58 @@
...
+  在此改进之前，如果从旧版本（4.2及之前）的 EMQX 导出数据，并导入到4.4版本，内置认证部分的数据
```
Comment (terry-xiaoyu) [inline zh_CN/changes/changes-ee-4.4.md]:
```suggestion
  在此改进之前，如果从旧版本（4.2及之前）导入到4.4版本，内置认证部分的数据
```

File: zh_CN/changes/changes-ee-4.4.md [hunk 20]
```diff
@@ -1,12 +1,58 @@
...
+  可能会因为缺少认证类型而被忽略，但用户无法从日志中清楚地获知失败原因。
+  在此修改之后，将会提示用户改用命令行工具做数据导入，并指定认证类型：
```
Comment (terry-xiaoyu) [inline zh_CN/changes/changes-ee-4.4.md]:
```suggestion
  在此修改之后，将会在 EMQX 日志中提示用户改用命令行工具做数据导入，并指定认证类型：
```

File: zh_CN/changes/changes-ee-4.4.md [hunk 21]
```diff
@@ -1,12 +1,58 @@
...
+
+### 修复
+
+- 在 EMQX 启动时迁移“内置认证”模块的 ACL 表 [#1776](https://github.com/emqx/emqx-enterprise/pull/1776)。
+  在此改进之前，如果通过拷贝 `data/mnesia/<node-name>` 目录的方式，将数据从 4.3 版本迁移到 4.4 版本，迁移完成后，
+  当通过 Dashboard 查看“内置认证”模块时，由于 ACL 表未迁移到新版格式，将出现 500 错误。
+  此问题仅在模块处于禁用状态时才会出现，并且可以通过手动启用模块来解决。
+  此修复后，EMQX 将在启动时尝试迁移 ACL 表，从而避免此问题。
+
+- 修复 IoTDB 动作的计数统计问题 [#1777](https://github.com/emqx/emqx-enterprise/pull/1777).
```
Comment (terry-xiaoyu) [inline zh_CN/changes/changes-ee-4.4.md]:
```suggestion
- 修复 IoTDB 动作的计数统计错误的问题 [#1777](https://github.com/emqx/emqx-enterprise/pull/1777).
```

File: zh_CN/changes/changes-ee-4.4.md [hunk 22]
```diff
@@ -1,12 +1,58 @@
...
+  在此改动之前，如果所有物理量（Measurement）都为 null，IoTDB 会将其忽略，不插入任何数据，
+  但是，IoTDB 仍会返回 200 OK，导致 IoTDB 动作递增发送成功计数。
+  此次修复后，当所有物理量都为 null 时，IoTDB 动作将丢弃此请求，并递增发送失败计数。
+
+- 修复无法使用带有换行符的 TDEngine SQL 语句创建规则的问题 [#1778](https://github.com/emqx/emqx-enterprise/pull/1778)。
```
Comment (terry-xiaoyu) [inline zh_CN/changes/changes-ee-4.4.md]:
```suggestion
- 修复当 TDEngine SQL 语句带有换行符时创建规则失败的问题 [#1778](https://github.com/emqx/emqx-enterprise/pull/1778)。
```

File: zh_CN/changes/changes-ee-4.4.md [hunk 23]
```diff
@@ -1,12 +1,58 @@
...
+  ```
+  INSERT INTO ${devid}
+  USING
+    tsdb.profit
+  TAGS
+    ('${custid}', '${devid}')
+  VALUES (${ts}, ${value})
+  ```
+
+- 修复 HTTP 接口 `/load_rebalance/:node/start` 返回的错误信息没有正确格式化的问题 [#1779](https://github.com/emqx/emqx-enterprise/pull/1779)。
```
Comment (terry-xiaoyu) [inline zh_CN/changes/changes-ee-4.4.md]:
```suggestion
- 修复 HTTP 接口 `/load_rebalance/:node/start` 返回的错误信息没有正确编码的问题 [#1779](https://github.com/emqx/emqx-enterprise/pull/1779)。
```

File: zh_CN/changes/changes-ee-4.4.md [hunk 24]
```diff
@@ -1,12 +1,58 @@
...
+
+- 修复 RocketMQ 生产者进程泄漏的问题 [rocketmq-client-erl#24](https://github.com/emqx/rocketmq-client-erl/pull/24)。
```
Comment (terry-xiaoyu) [inline zh_CN/changes/changes-ee-4.4.md]:
```suggestion
- 修复 EMQX 中 RocketMQ 客户端进程泄漏的问题 [rocketmq-client-erl#24](https://github.com/emqx/rocketmq-client-erl/pull/24)。
```

File: zh_CN/changes/changes-ee-4.4.md [hunk 25]
```diff
@@ -1,12 +1,58 @@
...
+  在此修复之前，EMQX 的 RocketMQ 客户端会周期性获取 RocketMQ 的节点信息，并检查节点信息是否有更新，
+  如果是则更新或者添加生产者进程。由于对比节点信息的方法有问题，某些情况下 RocketMQ 客户端会新建过多的生产者进程。
+  此次修复后，RocketMQ 客户端的生产者进程将会更加稳定。
```
Comment (terry-xiaoyu) [inline zh_CN/changes/changes-ee-4.4.md]:
```suggestion
```

File: zh_CN/changes/changes-ee-4.4.md [hunk 26]
```diff
@@ -1,12 +1,57 @@
...
+## e4.4.17
+
+*发布日期: 2023-04-13*
+
+### 增强
+
+- 改进从旧版本（4.2或更早）导入数据时的告警日志 [#1775](https://github.com/emqx/emqx-enterprise/pull/1775)。
+  在此改进之前，如果从旧版本（4.2及之前）导入到4.4版本，内置认证部分的数据
+  可能会因为缺少认证类型而被忽略，但用户无法从日志中清楚地获知失败原因。
+  在此修改之后，将会在 EMQX 日志中提示用户改用命令行工具做数据导入，并指定认证类型：
```
Comment (LenaLenaPan) [inline zh_CN/changes/changes-ee-4.4.md]:
```suggestion
  在此修改之后，将会在 EMQX 日志中提示用户改用命令行工具进行数据导入，并指定认证类型：
```

**Pull request merged**